src/components/picker/testing/index.tsx

```diff
@@ -1,35 +1,109 @@
+}
 
 @Component({
-	tag: "smoothly-picker-tester",
+	tag: "smoothly-picker-demo",
 	styleUrl: "style.css",
 	scoped: true,
 })
-export class SmoothlyPickerTester {
+export class SmoothlyPickerDemo {
+	@State() readonly = false
+	private users: Record<string, string | undefined> = {
+		"giovani@rocket.com": "giovani doe",
+		"jessie@rocket.com": "jessie doe",
+		"james@rocket.com": "james doe",
+	}
 	@State() data = {
 		message: "hello world",
 		emails: ["giovani@rocket.com", "jessie@rocket.com", "james@rocket.com"],
 	}
+	inputHandler(event: CustomEvent<Data>) {
+		this.data = { ...this.data, ...event.detail }
+		console.log(this.data)
+	}
+	@Listen("smoothlyFormSubmit")
+	submitHandler(event: CustomEvent<Data>) {
+		console.log("submitted", event.detail)
+	}
 	render() {
 		return (
-			<smoothly-form
-				looks="grid"
-				onSmoothlyFormInput={e => (this.data = { ...this.data, ...e.detail })}
-				onSmoothlyFormSubmit={e => console.log("submit", { ...this.data, ...e.detail })}>
-				<smoothly-input type="text" name="message" value={this.data.message}>
-					Message
-				</smoothly-input>
-				<smoothly-picker name="type" multiple mutable>
-					<span slot="label">Emails</span>
-					<span slot="search">Search</span>
-					{this.data.emails.map(email => (
-						<smoothly-picker-option value={email} selected>
-							{email}
+			<Host>
+				<smoothly-button onClick={() => (this.readonly = !this.readonly)}>
+					{this.readonly ? "current readonly" : "current writeable"}
+				</smoothly-button>
+				<h5>Controlled input</h5>
+				<smoothly-form looks="line" onSmoothlyFormInput={e => this.inputHandler(e)}>
+					<smoothly-input readonly={this.readonly} name="message" value={this.data.message}>
+						Message
+					</smoothly-input>
+					<smoothly-picker name="emails" mutable multiple readonly={this.readonly} validator={validateEmail}>
+						<span slot="label">Emails</span>
+						<span slot="search">Search</span>
+						<smoothly-icon size="tiny" slot="display" name="person-add-outline" />
+						{this.data.emails.map(email => (
+							<smoothly-picker-option value={email} selected search={[this.users[email] ?? []].flat()}>
+								<span>{this.users[email]}</span>
+								<span slot="label">{email}</span>
+								<smoothly-icon size="tiny" slot="display" name="person-outline" />
+							</smoothly-picker-option>
+						))}
+					</smoothly-picker>
+					<smoothly-submit size="icon" slot="submit">
+						<smoothly-icon name="checkmark-circle" />
+					</smoothly-submit>
+					<smoothly-input-clear type="form" color="danger" fill="solid" slot="clear">
+						Clear
+					</smoothly-input-clear>
+				</smoothly-form>
+				<h5>uncontrolled inputs</h5>
+				<smoothly-form onSmoothlyFormSubmit={e => console.log("submitted", e.detail)} looks="line">
+					<smoothly-input name="purpose" type="text">
+						Purpose
+					</smoothly-input>
+					<smoothly-picker multiple mutable name="emails" validator={validateEmail}>
```

Comment:
This picker does not add space between the emails on the label. Not sure where the problem is with that.

Reply:
completely missed that. now manually adding a span so the CSS gap rule is effective. more manual DOM manipulation :upside_down_face: